8th PR: Implement Role Selection Flow + Signup & Homepage UI Enhancements

FrankFurter123 opened the pull request:
## **Implement Role Selection Flow + Signup & Homepage UI Enhancements**  

---

## **Overview**  
### **What does this PR do?**  
This PR adds a complete **pre-signup role selection flow** and improves the **signup form** and **homepage UI** to support the updated onboarding experience. These changes are critical for enabling the role-based questionnaire flow and improving navigation clarity.

---

## **Key Features & Changes**  
### **What has been added or changed?**  
✔ **Core Feature Implementation**:  
- Added `RoleSelect.jsx` to prompt users to choose their role before sign-up.  
- Supports routing selected role into the Sign Up form via `navigate` state.

✔ **UI/UX Enhancements**:  
- Updated homepage with new images and section layout for visual clarity.  
- Refined "Join Now" CTA section text to reflect actual user flow.  
- Signup form now includes a password visibility toggle and full country list.  
- Added "Back to Home" buttons across Login, SignUp, and Role Select pages.

✔ **Navigation & Routing Fixes**:  
- "Join Now" button now correctly routes to `/RoleSelect`.  
- Cleaned up navigation by removing the unused Resources page.

✔ **New Components/Files**:  
- `RoleSelect.jsx` for role pre-selection  
- Updated `SignUp.jsx`, `Login.jsx`, and homepage components accordingly.

---

## **Why This Is Needed**  
### **What problem does this solve?**  
This setup is essential to support **role-based onboarding** and ensure users land on the correct flow from the start. It removes guesswork from the registration process and prepares us for role-aware backend integration and the post-login questionnaire modal.

---

## **Related Issue**  
🔗 **This PR addresses** #168 

---

## **Implementation Details**  
### **How was this feature developed?**  
- RoleSelect uses React Router’s `useNavigate` with state passing to SignUp.  
- Password field toggle handled with local `useState` logic.  
- Country list generated from a static array for dropdown.  
- Homepage image and layout updates use standard HTML/CSS with mobile-responsiveness in mind.  
- Routing logic fixed in `App.jsx`.

---

## **Design Documentation (If Applicable)**  
🖼️

- Screenshots of updated UI for RoleSelect
<img width="1512" alt="Screenshot 2025-04-20 at 14 47 02" src="https://github.com/user-attachments/assets/4ed48366-f285-4f43-bd2d-24b18c6df60b" />

- SignUp
<img width="1512" alt="Screenshot 2025-04-20 at 14 47 38" src="https://github.com/user-attachments/assets/e4bba968-45ea-4608-bea4-aac763080122" />

- Countries
<img width="1512" alt="Screenshot 2025-04-20 at 14 47 30" src="https://github.com/user-attachments/assets/28c0e4d7-bdf2-4d34-aebc-9c89e2538df2" />

-  Homepage section 1
<img width="1512" alt="Screenshot 2025-04-20 at 14 46 26" src="https://github.com/user-attachments/assets/1be1b007-3f5d-4591-9042-bb60bb03ed16" />

-  Homepage section 2
<img width="1512" alt="Screenshot 2025-04-20 at 14 46 42" src="https://github.com/user-attachments/assets/4257ced2-03a2-474b-8d63-13bc699d53b4" />


---

## **How to Test This PR**  
### **Step-by-step testing guide**  
1️⃣ Go to homepage → Click "Join Now" → It should take you to RoleSelect.  
2️⃣ Select a role → You should be navigated to Sign Up with the correct role.  
3️⃣ On Sign Up form:  
   - Test the password visibility toggle.  
   - Ensure the country dropdown loads and works.  
4️⃣ Check “Back to Home” buttons on all login/signup-related pages.  
5️⃣ Verify updated homepage layout and image alignment.  
6️⃣ Confirm Resources page is no longer accessible in nav or route.

---

## **Files Added/Modified**  
📂 **List of important files changed in this PR**  

| **File Name**            | **Description**                        |
|--------------------------|----------------------------------------|
| `RoleSelect.jsx`         | New role selection screen              |
| `SignUp.jsx`             | Updated with role param, toggle, country |
| `Login.jsx`              | Added back-to-home button              |
| `Home.jsx`               | Updated CTA flow and images            |
| `App.jsx`                | Routing updated for RoleSelect         |
| `Nav.jsx`                | Removed link to deprecated Resources   |
| `Resources.jsx` (deleted) | Removed page                          |

---

## **AI Code Usage**  
🤖 **Was AI-generated code used in this PR?**  

- [x] No (_All code was manually implemented._)

---

## **Checklist for Reviewers**  
**Things to check before approving this PR:**  
- [x] RoleSelect page shows up and routes correctly.  
- [x] SignUp form updates based on selected role.  
- [x] Join Now CTA leads to the proper page.  
- [x] Back navigation works across login/signup/role select.  
- [x] UI is responsive across screen sizes.

---

## **Next Steps & Future Enhancements**  
### **What improvements can be made after merging?**  
- Connect selected role from RoleSelect to backend user profile creation.  
- Implement validation feedback on SignUp form.  
- Add animations/transitions between pages.  
- Offer role re-selection/editing after login from settings.

---

### **Final Notes**  
Thanks for reviewing! Let me know if anything looks off or can be improved.

AbdulAlharbi reviewed (approved):
### Peer Review: Great Job on This PR!

Hey Frank 👋 — just finished reviewing your PR titled **"Implement Role Selection Flow + Signup & Homepage UI Enhancements"**. Here’s my feedback:

---

## What Looks Great

- The **core functionality** of role-based onboarding has been implemented cleanly and works as expected. I tested the full flow — from the homepage CTA through role selection to signup — and it's seamless.
- You’ve done a stellar job with **UI/UX consistency**. The new visual updates feel polished, intuitive, and responsive across screen sizes.
- Love the addition of **country selection** and **password visibility toggle** — thoughtful touches that boost usability.
- The **navigation fixes** (e.g., Join Now routing, Back buttons, and removal of Resources page) contribute to a much smoother user experience.
- Your **PR description is exemplary**: clear summary, detailed breakdown, testing steps, and even image references. It made reviewing this super efficient.
- You've properly linked to the **related issue** (#168) and clarified how this sets up future role-specific flows — nice foresight!

---

## Suggestions for Improvement (After MVP) 

> _Nothing blocking here — just future-forward suggestions._

- ⚠️ **Design Diagram Missing**: Since this introduces a new flow (`RoleSelect → SignUp`), adding a simple **use case or sequence diagram** would help document this logic.
- Consider making the country dropdown searchable in future — this will improve user experience, especially for long lists.
- A next-level polish idea: Add subtle **page transition animations** between the onboarding steps (Role → Signup) to enhance continuity.

---

## Tested This Feature

Here’s what I verified during testing:

- ✅ Homepage “Join LessonConnect” routes to `RoleSelect`.
- ✅ Role buttons correctly pass data to the `SignUp` form.
- ✅ Password visibility toggle works as expected.
- ✅ Country list populates and is selectable.
- ✅ “Back to Home” buttons function across all related pages.
- ✅ UI adapts well to various screen sizes.
- ✅ Navigation no longer links to the deprecated Resources page.

All behavior is correct, and no bugs encountered ✅

---

## Next Steps

- Let’s get a **UML diagram** added (even a lightweight one) to lock this in for design documentation completeness.
- Once backend integration is in place, make sure the selected role persists in the user profile creation.

---

### Final Thoughts 

Excellent job with this PR, Frank!  
**✅ Approved!**  
It’s thoughtful, user-focused, and technically sound — and it paves the way for robust, role-based logic in our platform. Let me know if you want help sketching a quick diagram to pair with the PR.

Looking forward to seeing this merged and driving the next stage of onboarding! 
-_Abdulrahman_

FrankFurter123 commented:
Hey @AbdulAlharbi  — thank you so much for the review!

I really appreciate you testing the flow end-to-end and calling out those UI/UX touches. It means a lot to hear that the experience felt seamless and polished — that was definitely the goal!  Also glad the PR description helped streamline your review process — I’ve been trying to make sure they’re clear and comprehensive.

Totally agree with your suggestions, especially around adding a searchable country dropdown and some subtle transitions to elevate the flow. And you're right — now that the role selection step is in place, it’s a great time to lock in the design logic for future reference.

Here’s the UML diagram I was working on! 🧩  


This UML diagram illustrates the key components and flow introduced in this PR:
![UML diagram](https://github.com/user-attachments/assets/6d6a2953-7625-40c6-a01e-ab980f8d641c)

- **HomePage.jsx**: Initiates the onboarding flow with a "Join Now" CTA that routes users to the `RoleSelect` component.  
- **RoleSelect.jsx**: Presents users with role options and uses React Router’s `navigate()` to pass the selected role to `SignUp.jsx`.  
- **SignUp.jsx**: Receives the role state, handles form inputs (including country dropdown and password visibility toggle).  
- **Login.jsx**: Also updated with consistent UI and navigation.  
- **App.jsx**: Manages updated routes and integration of new components.  
- **Nav.jsx**: Cleaned up to remove the deprecated `Resources` link.  
- **Resources.jsx**: Fully removed from the codebase.

The diagram highlights method-level responsibilities and navigation links between the components to support the new role-based onboarding experience.

Thank you again Abdul for the approval of this PR!

ashley-arellano reviewed (approved):
### 🔍 Peer Review: Great Job on This PR!

Hey Frank 👋 — just finished reviewing your PR titled **"Implement Role Selection Flow + Signup & Homepage UI Enhancements"**. Here’s my feedback:

----------

## ✅ What Looks Great

-   The role selection flow is smooth. The `RoleSelect.jsx` screen works well and cleanly passes state to `SignUp.jsx`.
    
-   Nice touch adding the password visibility toggle and complete country dropdown. It makes it convienent for the user.
    
-   Also, you added some amazing UI improvements. Images and content are more pleasant to look at now.
    
-   Great call removing the unused Resources page and cleaning up navigation.
    
-   Your PR description is extremely detailed, with screenshots, implementation notes, and a full test guide.
    

----------

## 💡 Suggestions for Improvement (if any)

> _These are ideas for future improvements or things to address before merging._
    
-   If the user navigates directly to the SignUp page (bypassing RoleSelect), maybe include a fallback/default role or redirect logic. This could be added in the future.
    

----------

## 🧪 Tested This Feature

I ran the following test steps:

-   ✅ Clicked "Join Now" on homepage → routed to `RoleSelect`.
    
-   ✅ Selected both roles which were passed properly to `SignUp`.
    
-   ✅ Signup form UI elements work (toggle, country list, back-to-home).
    
-   ✅ Tried navigating to deprecated `/resources` — correctly inaccessible.
    

Everything behaved as expected 🎉

----------

## 🔄 Next Steps

-   Connect the selected role to the backend for actual account creation logic.
 
    

----------

### 🚀 Final Thoughts

Fantastic work! This PR adds a ton of value and improves the onboarding flow significantly. Great job : D

FrankFurter123 commented:
Hey @ashley-arellano — thank you so much for your review! 

Really appreciate you calling out the improvements — especially the UX touches like the password toggle and layout polish. Glad the flow from homepage → role select → signup felt smooth on your end!

Great catch on the SignUp fallback logic — definitely planning to handle direct access in the next PR if time allows, either by redirecting to RoleSelect or applying a default role with a warning.

Thanks again for testing everything thoroughly and for all the support!  I will go ahead and merge this now!

Thank you again!
